sys/riscv/riscv/unwind.c

```diff
@@ -42,16 +42,21 @@ __FBSDID("$FreeBSD$");
 int
 unwind_frame(struct unwind_state *frame)
 {
-	uint64_t fp;
+	uintptr_t fp;
 
 	fp = frame->fp;
 
 	if (!INKERNEL(fp))
 		return (-1);
 
 	frame->sp = fp;
-	frame->fp = *(uint64_t *)(fp - 16);
-	frame->pc = *(uint64_t *)(fp - 8) - 4;
+	frame->fp = ((uintptr_t *)fp)[-2];
+#ifdef __CHERI_PURE_CAPABILITY__
+	/* The saved return address is sealed so cannot be modified. */
```

Comment:
Although if you clear the tag you can... perhaps an argument in favour of the Morello approach. So if you really want the right PC (compression aside) you can manually do a `cheri_cleartag` (or even just get a NULL-derived address if that's all you need).

Reply:
I used a vaddr_t cast to make it null derived and this worked fine for DDB unwinding at least.  We only use the virtual address to either print directly or to lookup symbol names.